hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/conf/TestOzoneConfiguration.java

```diff
@@ -0,0 +1,145 @@
+      startConfig(out);
+      appendProperty(out, "hadoop.tags.system", "YARN,HDFS,NAMENODE");
+      appendProperty(out, "hadoop.tags.custom", "MYCUSTOMTAG");
+      appendPropertyByTag(out, "dfs.cblock.trace.io", "false", "YARN");
+      appendPropertyByTag(out, "dfs.replication", "1", "HDFS");
+      appendPropertyByTag(out, "dfs.namenode.logging.level", "INFO",
+          "NAMENODE");
+      appendPropertyByTag(out, "dfs.random.key", "XYZ", "MYCUSTOMTAG");
+      endConfig(out);
+
+      Path fileResource = new Path(coreDefault.getAbsolutePath());
+      conf.addResource(fileResource);
+      assertEq(conf.getAllPropertiesByTag("MYCUSTOMTAG")
+          .getProperty("dfs.random.key"), "XYZ");
+    }
+
+    try (BufferedWriter out = new BufferedWriter(new FileWriter(coreSite))) {
+      startConfig(out);
+      appendProperty(out, "dfs.random.key", "ABC");
+      appendProperty(out, "dfs.replication", "3");
+      appendProperty(out, "dfs.cblock.trace.io", "true");
+      endConfig(out);
+
+      Path fileResource = new Path(coreSite.getAbsolutePath());
+      conf.addResource(fileResource);
+    }
+
+    // Test if values are getting overridden even without tags being present
+    assertEq("3", conf.getAllPropertiesByTag("HDFS")
+        .getProperty("dfs.replication"));
+    assertEq("ABC", conf.getAllPropertiesByTag("MYCUSTOMTAG")
+        .getProperty("dfs.random.key"));
+    assertEq("true", conf.getAllPropertiesByTag("YARN")
+        .getProperty("dfs.cblock.trace.io"));
+  }
+
+  private void appendProperty(BufferedWriter out, String name, String val)
+      throws IOException {
+    this.appendProperty(out, name, val, false);
+  }
+
+  private void appendProperty(BufferedWriter out, String name, String val,
+                              boolean isFinal) throws IOException {
+    out.write("<property>");
+    out.write("<name>");
+    out.write(name);
+    out.write("</name>");
+    out.write("<value>");
+    out.write(val);
+    out.write("</value>");
+    if (isFinal) {
+      out.write("<final>true</final>");
+    }
+    out.write("</property>\n");
+  }
+
+  private void appendPropertyByTag(BufferedWriter out, String name, String val,
+                                   String tags) throws IOException {
+    this.appendPropertyByTag(out, name, val, false, tags);
+  }
+
+  private void appendPropertyByTag(BufferedWriter out, String name, String val,
+                                   boolean isFinal,
+                                   String tag) throws IOException {
+    out.write("<property>");
+    out.write("<name>");
+    out.write(name);
+    out.write("</name>");
+    out.write("<value>");
+    out.write(val);
+    out.write("</value>");
+    if (isFinal) {
+      out.write("<final>true</final>");
+    }
+    out.write("<tag>");
+    out.write(tag);
+    out.write("</tag>");
+    out.write("</property>\n");
+  }
+
+  private static void assertEq(Object a, Object b) {
```

Comment:
Instead we can directly use assertEquals, we don't need to do System.out.println